### ISSUE TITLE
Crash on rename --cursor=start

### ISSUE THREAD

**Issue by Sonico98** [sxyazi/yazi#573]
### What system are you running Yazi on?

Linux Wayland

### What terminal are you running Yazi in?

Kitty 0.31.0

### Yazi version

yazi 0.2.2 (a19e263 2024-01-24)

### Did you try the latest main branch to see if the problem has already been fixed?

Tried, but the problem is still present

### Describe the bug

Trying to rename a file using `rename --cursor=start` crashes yazi if the file name contains 48 characters or more.

### Expected Behavior

To be able to rename the file. Note that using rename --cursor=before_ext works fine.

### To Reproduce

1- Create a file with the following name: AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA
2- Use keybindings to rename that file using `rename --cursor=start`
3- yazi crashes

Output with RUST_BACKTRACE=full (it outputs the same as RUST_BACKTRACE=1)
```
Backtrace (most recent call first):
  File "<unknown>", line 0, in __libc_start_main

The application panicked (crashed).
  begin <= end (11 <= 0) when slicing `AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA`
in /usr/src/debug/yazi-git/yazi/yazi-core/src/input/snap.rs, line 61
thread: main
```

### Configuration

The following keymap.toml is all needed to replicate the crash, assuming the file with the offending file name already exists in the current working directory.
```
[manager]
keymap = [
	{ on = [ "k" ], exec = "arrow -1", desc = "Move cursor up" },
	{ on = [ "j" ], exec = "arrow 1",  desc = "Move cursor down" },
	{ on = [ "r", "a" ], exec = "rename --cursor=before_ext",         desc = "Rename a file or directory" },
	{ on = [ "r", "A" ], exec = "rename --cursor=start",              desc = "Rename a file or directory, from the start of the file" },
	{ on = [ "r", "e" ], exec = "rename --empty=stem --cursor=start", desc = "Rename a file or directory (clear existing file name)" },
]
```

**Comment by sxyazi** [comment]
Thanks for the reporting, should be fixed by https://github.com/sxyazi/yazi/pull/575

**Comment by Sonico98** [comment]
> Thanks for the reporting, should be fixed by #575

Thanks for the quick fix and the inmense amount of work you've done on this file manager! I love it!

**Comment by sxyazi** [comment]
Ohh thank you! :3
